LEAF_Request_Portal/templates/subindicators.tpl

```diff
@@ -224,35 +223,56 @@
-                        <!--{assign var='found' value=true}-->
-                    <!--{/if}-->
-                <!--{/foreach}-->
-                <!--{if !$found}-->
-                    <option value="<!--{$option|sanitize}-->"><!--{$option|sanitize}--></option>
-                <!--{/if}-->
-            <!--{/foreach}-->
-                </select></span>
-                <input type="hidden" id="<!--{$indicator.indicatorID|strip_tags}-->_selected" name="<!--{$indicator.indicatorID|strip_tags}-->_selected" value="" />
-                <script>
+            <select multiple
+                id="<!--{$indicator.indicatorID|strip_tags}-->"
+                name="<!--{$indicator.indicatorID|strip_tags}-->_multiselect[]"
+                style="display:none">
+            </select>
+
+            <script>
+                if (typeof indicatorInfo === 'undefined') {
+                    var indicatorInfo = {}; //needs to be var.
+                }
+                indicatorInfo[<!--{$indicator.indicatorID|strip_tags}-->] = {
+                    indOptions: <!--{$indicator.options|json_encode}--> || [],
+                    indValues: Array.isArray(<!--{$indicator.value|json_encode}-->) ? 
+                               <!--{$indicator.value|json_encode}--> :      //new serialized array format
+                               '<!--{$indicator.value}-->'.split(/,(?!\s)/) //old concat string format compatible (needed for default vals)
+                }
                 $(function() {
-                    $('#<!--{$indicator.indicatorID|strip_tags}-->').chosen({width: '80%'});
-                    $('#<!--{$indicator.indicatorID|strip_tags}-->_chosen .chosen-choices').css('border-radius', '6px');
-                    // Hidden Value for array of items in _selected to export to POST
-                    let hiddenValue = $('#<!--{$indicator.indicatorID|strip_tags}-->_chosen .chosen-choices')[0].innerText;
-                    $('#<!--{$indicator.indicatorID|strip_tags}-->_selected').val(hiddenValue.split("\n"));
-                    // Change function for updating array on each selection or deselection
-                    $('#<!--{$indicator.indicatorID|strip_tags}-->').on('change', function() {
-                        setTimeout(function() {
-                            hiddenValue = $('#<!--{$indicator.indicatorID|strip_tags}-->_chosen .chosen-choices')[0].innerText;
-                            $('#<!--{$indicator.indicatorID|strip_tags}-->_selected').val(hiddenValue.split("\n"));
-                        }, 500);
-                    });
+                    for (let i in indicatorInfo) {
+                        const elSelect = document.getElementById(i);
+                        if (elSelect !== null && elSelect.multiple === true && elSelect.getAttribute('data-choice') !== 'active') {
+                            function decodeHTMLEntities(str) {
```

Comment:
Why is this function buried in multiple levels?  This looks like it should be on the root level.